sherpa/astro/data.py

```diff
@@ -247,39 +247,37 @@ def make_metadata(header, items):
     return meta
 
 
-def _extract_fields(obj, stop, summary, open_block=True):
-    """Extract the fields up until the stop field.
+def _extract_fields(obj, summary):
+    """Extract the "column" fields.
+
+    Write out the _fields values (that are not None) for the Data
+    object. The _extra_fields values are ignored, as thet are assumed
+    to be icluded in separate "metadata" section.
 
     Parameters
     ----------
     obj : Data instance
-        It has to have a _fields attribute
-    stop : str
-        The attribute at which to stop (and is not included).
     summary : str
         The label for the details tab.
-    open_block : bool, optional
-        Is the details tab open or closed?
 
     Returns
     -------
     html : str
         The HTML for this section.
+
     """
 
     meta = []
-    for f in obj._fields[1:]:
-        if f == stop:
-            break
-
+    fields = [f for f in obj._fields if f != 'name']
```

Comment:
To be honest we could include the name as well, it would just be redundant. I do think we are possibly missing something here - `_extra_fields` helps out for the purposes of this PR / bug, but maybe there's some other way to look at what we need for the various options - e.g. for __str__, notebook output, serialization (this PR), anything else? - that would indicate what we really need.

Reply:
I had that thought; and in fact I have a similar thought while working on the plotting code. I'm not sure an inline comment on a fairly targeted PR is the best place to discuss a larger picture like that, but I do have some text on that that was supposed to go into the description of one of the next plotting PRs, I'll turn that into a new issue (give me an hour) and then we can discuss there and come back here after that?

Reply:
kewl